modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java

```diff
@@ -399,9 +408,14 @@ public String toVarName(String name) {
         // remove the rest
         name = sanitizeName(name);
 
-        // camelize (lower first character) the variable name
-        // pet_id => petId
-        name = camelize(name, LOWERCASE_FIRST_LETTER);
+        if (useLowerCamelCase) {
+            //to camelize it correctly it needs to be lower cased
+            name = camelize(name.toLowerCase(), LOWERCASE_FIRST_LETTER);
```

Comment:
I think this is a problem, this results in all lowercase for var names.

Reply:
Ah yes you're right.
Sorry I'm not so into the code atm. Is there a better way to only detec enum cases?